### PR TITLE
FilePathUtils.joinZipPathSegments error on windows

### LINKED ISSUE #1592: FilePathUtils.joinZipPathSegments error on windows
On windows the `joinZipPathSegments` of the `FilePathUtils` does not work properly on windows. This is because files are not seperated by `/`, but rather by `\`. A hotfix was provided in #1591, but a proper solution is needed for the future.

### PR DESCRIPTION
**Pull request by TwoOfTwelve**
The file creation now uses the path api instead of manual concatenation. I cannot test it on windows. So it would be great if someone could confirm it also works on windows.

Fixes #1592

### CODE REVIEW

**Comment by Kr0nox** [comment]
This does not generate the expected report with the single submission files, so just a folder with files.
Input folder:
```
Folder
└Sub1.java
└Sub2.java
```

We would expect the report zip be look like this:
```
files
 └Sub1.java (This is a folder)
 |  └ Sub1.java
 └Sub2.java (This is a folder)
 |  └ Sub2.java
Json Files
```

But the zip looks like this:
```
files
 └Sub1.java (This is a file)
 └Sub2.java (This is a file)
Json Files
```

**Comment by Kr0nox** [comment]
> Also fixes https://github.com/jplag/JPlag/issues/1610

I would suggest moving this to a diferent PR, since it is very independent of the other issue

File: core/src/main/java/de/jplag/reporting/FilePathUtil.java [hunk 1]
```diff
@@ -28,23 +27,26 @@ public static String getRelativeSubmissionPath(File file, Submission submission,
...
-     * Joins logical paths using a slash. This method ensures, that no duplicate slashes are created in between.
-     * @param left The left path segment
-     * @param right The right path segment
-     * @return The joined paths
-     */
-    public static String joinZipPathSegments(String left, String right) {
-        String rightStripped = right;
-        while (rightStripped.startsWith(ZIP_PATH_SEPARATOR) || rightStripped.startsWith(WINDOWS_PATH_SEPARATOR)) {
-            rightStripped = rightStripped.substring(1);
+    public static Path forceRelativePath(Path path) {
```
Comment (Kr0nox) [inline core/src/main/java/de/jplag/reporting/FilePathUtil.java:30]: JDoc

File: core/src/main/java/de/jplag/reporting/FilePathUtil.java [hunk 2]
```diff
@@ -28,23 +27,26 @@ public static String getRelativeSubmissionPath(File file, Submission submission,
...
+            return Path.of("/").relativize(path);
         }
+        return path;
+    }
 
-        String leftStripped = left;
-        while (leftStripped.endsWith(ZIP_PATH_SEPARATOR) || leftStripped.startsWith(WINDOWS_PATH_SEPARATOR)) {
-            leftStripped = leftStripped.substring(0, leftStripped.length() - 1);
-        }
+    public static Path createRelativePath(String path) {
```
Comment (Kr0nox) [inline core/src/main/java/de/jplag/reporting/FilePathUtil.java:37]: JDoc

File: core/src/main/java/de/jplag/reporting/FilePathUtil.java [hunk 3]
```diff
@@ -28,23 +27,26 @@ public static String getRelativeSubmissionPath(File file, Submission submission,
...
+        return forceRelativePath(Path.of(path));
+    }
 
-        return leftStripped + ZIP_PATH_SEPARATOR + rightStripped;
+    public static String pathAsZipPath(Path path) {
```
Comment (Kr0nox) [inline core/src/main/java/de/jplag/reporting/FilePathUtil.java:41]: JDoc

**Review by Kr0nox — CHANGES_REQUESTED** [review]
Except for some JDoc this looks good.

Tested on windows

**Review by Kr0nox — APPROVED** [review]
Tested on windows. Works with:
- [x] Single Submission Folder
- [x] Multiple Submission Folders
- [x] Single Folder of Submission Foles
- [x] Multiple Folders of submission files (appart from #1610

**Review by tsaglam — APPROVED** [review]

**Pull request merged**